_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,177 @@
+</iframe>
+```
+
+#### Pass example 4
+
+Multiple `iframe` elements in the document having different `aria-labelledby` descriptions as accessible name.
+
+```html
+<div id="desc-for-title">List of Contributors</div>
+<iframe aria-labelledby="desc-for-title" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<div id="desc-for-title1">List of Reviewers</div>
+<iframe aria-labelledby="desc-for-title1" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+### Failed
+
+#### Fail example 1
+
+Multiple frames having the same `title` within a given document context.
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Fail example 2
+
+Multiple frames having the same `aria-label` within a given document context.
+
+```html
+<iframe aria-label="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Fail example 3
+
+Multiple frames having the same name given by `title` and `aria-label` within a given document context
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+### Inapplicable
+
+#### Inapplicable example 1
+
+Accessible name provided is empty.
+
+```html
+<iframe title="" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="" src="../test-assets/SC4-1-2-iframe-unique-name-doc3.html">
+</iframe>
+```
+
+#### Inapplicable example 2
+
+Usage of `alt` attribute to describe content is not valid.
+
+```html
+<iframe alt="Some" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+```
+
+#### Inapplicable example 3
+
+Does not apply to non `iframe` elements.
+
+```html
+<object title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</object>
+<object aria-label="List of Contributors Clone" src="../test-assets/SC4-1-2-iframe-unique-name-doc3.html">
+</object>
+```
+
+#### Inapplicable example 4
+
+No accessible name is provided
+
+```html
+<iframe src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe src="../test-assets/SC4-1-2-iframe-unique-name-doc3.html">
+</iframe>
+```
+
+#### Inapplicable example 5
+
+Does not apply to `iframe` elements that are not exposed to assistive technologies.
+
+```html
+<iframe aria-hidden="true" title="Document One" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-hidden="true" aria-label="Document One" src="../test-assets/SC4-1-2-iframe-unique-name-doc3.html">
```

Comment:
Would it be useful to have a test case with one of these iframes exposed and one hidden?

Reply:
It would be useful to add it as a pass test case with 2 iframe(s) having same accessible name, but one of them as hidden. Doing that.